packages/stack/src/views/Header/Header.tsx

```diff
@@ -4,64 +4,56 @@ import HeaderSegment from './HeaderSegment';
 import { HeaderProps, HeaderTitleProps } from '../../types';
 import HeaderTitle from './HeaderTitle';
 
-export default class Header extends React.PureComponent<HeaderProps> {
-  render() {
-    const {
-      scene,
-      previous,
-      layout,
-      navigation,
-      styleInterpolator,
-    } = this.props;
-    const { options } = scene.descriptor;
-    const title =
-      typeof options.headerTitle !== 'function' &&
-      options.headerTitle !== undefined
-        ? options.headerTitle
-        : options.title !== undefined
-        ? options.title
-        : scene.route.name;
+export default React.memo(function(props: HeaderProps) {
```

Comment:
Can you explain why you use `React.memo` here? AFAIK there’s no need for it

Reply:
previously it was extending PureComponent, which is exact same as React.memo.

Reply:
Yes, but following my understanding it should work well w/o this wrapper

Reply:
yes, it's kind of a performance optimization. React.memo will shallow compare props, and won't re-render if nothing changed.

Reply:
it's a kind of performance optimization provided by PureComponent